distribution/openrct2.d.ts

```diff
@@ -1742,10 +1746,46 @@ declare global {
         readonly numVerticalFramesOverride: number;
     }
 
+    interface SceneryObject extends LoadedImageObject {
+        /**
+         * A list of scenery groups this object belongs to. This may not contain any
+         * scenery groups that contain this object by default. This is typically
+         * used for custom objects to be part of existing scenery groups.
+         */
+        readonly sceneryGroups: ObjectReference[];
+    }
+
+    /**
+     * Represents a reference to an object which may or may not be loaded.
+     */
+    interface ObjectReference {
+        /**
+         * The JSON identifier of the object.
+         * Undefined if object only has a legacy identifier.
+         */
+        identifier?: string;
+
+        /**
+         * The DAT identifier of the object.
+         * Undefined if object only has a JSON identifier.
+         */
+        legacyIdentifier?: string;
```

Comment:
We really should not be exposing this. It's the dat name not the identifier and it's the source of bugs already in the codebase where the name is assumed to be the identifier.

Reply:
How can plugins do anything with the majority of custom objects if we don't expose this?

Reply:
expose the full 16 byte identifier rather than the dat name.

Reply:
I would prioritise having it be consistent with `LoadedObject`.